### DISCUSSION TITLE
Minimum Distance Between Claims?

### DISCUSSION THREAD

**Discussion by Blzd87** [GriefPrevention/GriefPrevention#1754] [Ideas]
This feature would be great. 

If you could add a config setting for Minimum distance between claims?
for example in my SMP I have a rule that players need to keep a minimum distance of 100 blocks between claims to stop players from stunting anothers growth if they plan on increasing there base when there claim blocks increase.
As you know players dont like abiding by rules sometimes and like hassling others. and It would help admins not needing to always fix those issues. 

this is of course if this request is even possible with keeping server performance impact low as your plugin already has a small impact atm as it is. 

Please consider this request, Or any other ideas on how I can acheieve this?

Thanks.

**Comment by bigpresh** [comment]
I'm sure I've seen this raised as an issue before but couldn't find it with a quick search - but it's there somewhere I'm pretty sure.

I think searching for any nearby claims in any direction from any block of the proposed new claim could be fairly expensive, and could lead to confusion too.  (e.g. both corners you clicked are more than 100 blocks from another claim, but somewhere in the middle comes closer than 100 blocks to a narrow claim, etc - I think explaining/visualising instances like that would be more confusing than it's worth)

Also, if it prevents them from claiming nearer than 100 blocks "to stop players from stunting anothers growth if they plan on increasing there base when there claim blocks increase", then how would they be able to increase their claim later, if there's now a new claim 100 blocks from them and the plugin was preventing claims being closer than 100 blocks?

This is something that could probably be implemented as an addon by someone, but I don't think belongs in GP core.

**Reply by TinyTank800** [reply]
Depending on size of server I could see how that gets taxing but I would love to see this come to GP. I've ran into many instances where players claim right next to someone just to put shriekers and spawners there to annoy and damage players near the claim. I have also had people claim a small square inside someone's claims before. This gets really annoying to try and moderate distance or things allowed and would love to see some for of simple distance check.
